rust/stracciatella_c_api/src/c/fs.rs

```diff
@@ -0,0 +1,209 @@
+//! This module contains the C interface for [`stracciatella::fs`].
+//!
+//! [`stracciatella::fs`]: ../../stracciatella/fs/index.html
+
+use std::f64;
+use std::ptr;
+use std::time;
+
+use stracciatella::fs;
+
+use crate::c::common::*;
+use crate::c::misc::VecCString;
+
+/// A directory in the filesystem that is automatically deleted.
+/// The contents of the directory are deleted before the directory is deleted.
+pub struct TempDir {
+    inner: fs::TempDir,
+}
+
+/// Destroys the temporary directory.
+/// This is a best effort approach, in case of error the directory might not be deleted.
+/// Sets the rust error.
+#[no_mangle]
+pub extern "C" fn TempDir_destroy(tempdir: *mut TempDir) -> bool {
+    forget_rust_error();
+    let tempdir = from_ptr(tempdir);
+    if let Err(err) = tempdir.inner.close() {
+        remember_rust_error(format!("TempDir_destroy: {}", err));
+    }
+    no_rust_error()
+}
+
+/// Gets the path of the temporary directory.
+#[no_mangle]
+pub extern "C" fn TempDir_path(tempdir: *mut TempDir) -> *mut c_char {
+    let tempdir = unsafe_ref(tempdir);
+    let path = tempdir.inner.path();
+    let c_path = c_string_from_path_or_panic(path);
+    c_path.into_raw()
+}
+
+/// Creates a directory.
+/// Sets the rust error.
+#[no_mangle]
+pub extern "C" fn Fs_createDir(path: *const c_char) -> bool {
+    forget_rust_error();
+    let path = path_from_c_str_or_panic(unsafe_c_str(path));
+    if let Err(err) = fs::create_dir(&path) {
+        remember_rust_error(format!("Fs_createDir {:?}: {}", path, err));
+    }
+    no_rust_error()
+}
+
+/// Attempts to create a temporary directory inside the temporary directory of the operating system.
+/// Returns null on error.
+/// Sets the rust error.
+/// @see https://doc.rust-lang.org/std/env/fn.temp_dir.html
+#[no_mangle]
+pub extern "C" fn Fs_createTempDir() -> *mut TempDir {
+    forget_rust_error();
+    match fs::TempDir::new() {
+        Err(err) => {
+            remember_rust_error(format!("Fs_createTempDir: {}", err));
+            ptr::null_mut()
+        }
+        Ok(tempdir) => into_ptr(TempDir { inner: tempdir }),
+    }
+}
+
+/// Checks if the path exists.
+#[no_mangle]
+pub extern "C" fn Fs_exists(path: *const c_char) -> bool {
+    let path = path_from_c_str_or_panic(unsafe_c_str(path));
+    fs::metadata(&path).is_ok()
```

Comment:
You could just use `path.exists()` here.

Reply:
True, but I want to make it clear that it's touching the filesystem.